rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/geo/TbGpsGeofencingActionNodeTest.java

```diff
@@ -0,0 +1,269 @@
+        node = new TbGpsGeofencingActionNode();
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
+
+    private static Stream<Arguments> givenReportPresenceStatusOnEachMessage_whenOnMsg_thenVerifyOutputMsgType() {
+        DeviceId deviceId = new DeviceId(UUID.randomUUID());
+        long tsNow = System.currentTimeMillis();
+        long tsNowMinusMinuteAndMillis = tsNow - Duration.ofMinutes(1).plusMillis(1).toMillis();
+        return Stream.of(
+                // default config with presenceMonitoringStrategyOnEachMessage false and msgInside true
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false, new EntityGeofencingState(false, 0, false)), ENTERED),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false, new EntityGeofencingState(true, tsNow, false)), SUCCESS),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false, new EntityGeofencingState(true, tsNowMinusMinuteAndMillis, false)), INSIDE),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false, new EntityGeofencingState(true, tsNow, true)), SUCCESS),
+                // default config with presenceMonitoringStrategyOnEachMessage false and msgInside false
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false, new EntityGeofencingState(false, 0, false)), LEFT),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false, new EntityGeofencingState(false, tsNow, false)), SUCCESS),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false, new EntityGeofencingState(false, tsNowMinusMinuteAndMillis, false)), OUTSIDE),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false, new EntityGeofencingState(false, tsNow, true)), SUCCESS),
+                // default config with presenceMonitoringStrategyOnEachMessage true and msgInside true
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true, new EntityGeofencingState(false, 0, false)), ENTERED),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true, new EntityGeofencingState(true, tsNow, false)), INSIDE),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true, new EntityGeofencingState(true, tsNowMinusMinuteAndMillis, false)), INSIDE),
+                // default config with presenceMonitoringStrategyOnEachMessage true and msgInside false
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true, new EntityGeofencingState(false, 0, false)), LEFT),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true, new EntityGeofencingState(false, tsNow, false)), OUTSIDE),
+                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true, new EntityGeofencingState(false, tsNowMinusMinuteAndMillis, false)), OUTSIDE)
+        );
+    }
```

Comment:
Please move `new EntityGeofencingState(...` for each argument to next line so there will be no needs to scroll over to view the full argument definition.

```suggestion
    private static Stream<Arguments> givenReportPresenceStatusOnEachMessage_whenOnMsg_thenVerifyOutputMsgType() {
        DeviceId deviceId = new DeviceId(UUID.randomUUID());
        long tsNow = System.currentTimeMillis();
        long tsNowMinusMinuteAndMillis = tsNow - Duration.ofMinutes(1).plusMillis(1).toMillis();
        return Stream.of(
                // default config with presenceMonitoringStrategyOnEachMessage false and msgInside true
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false,
                        new EntityGeofencingState(false, 0, false)), ENTERED),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false,
                        new EntityGeofencingState(true, tsNow, false)), SUCCESS),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false,
                        new EntityGeofencingState(true, tsNowMinusMinuteAndMillis, false)), INSIDE),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, false,
                        new EntityGeofencingState(true, tsNow, true)), SUCCESS),
                // default config with presenceMonitoringStrategyOnEachMessage false and msgInside false
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false,
                        new EntityGeofencingState(false, 0, false)), LEFT),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false,
                        new EntityGeofencingState(false, tsNow, false)), SUCCESS),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false,
                        new EntityGeofencingState(false, tsNowMinusMinuteAndMillis, false)), OUTSIDE),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, false,
                        new EntityGeofencingState(false, tsNow, true)), SUCCESS),
                // default config with presenceMonitoringStrategyOnEachMessage true and msgInside true
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true,
                        new EntityGeofencingState(false, 0, false)), ENTERED),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true,
                        new EntityGeofencingState(true, tsNow, false)), INSIDE),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, true, true,
                        new EntityGeofencingState(true, tsNowMinusMinuteAndMillis, false)), INSIDE),
                // default config with presenceMonitoringStrategyOnEachMessage true and msgInside false
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true,
                        new EntityGeofencingState(false, 0, false)), LEFT),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true,
                        new EntityGeofencingState(false, tsNow, false)), OUTSIDE),
                Arguments.of(new GpsGeofencingActionTestCase(deviceId, false, true,
                        new EntityGeofencingState(false, tsNowMinusMinuteAndMillis, false)), OUTSIDE)
        );
    }
```